ayat/authentication_routes.py

```diff
@@ -1,191 +1,294 @@
+                'birth_date': user.birth_date,
+                'gender': user.gender,
+            }), 200
+
+
+    elif data['action'] == 'register':
+        try:
+            # checking if user exists or not
+            user_email = data['email']
+            user = User.query.filter_by(email=user_email).first()
+            if user is not None:
+                return jsonify({"status":  "1"})
+
+            user_phone = data['phone']
+            user = User.query.filter_by(phone_number=user_phone).first()
+            if user is not None:
+                return jsonify({"status":  "2"})
+
+            hashed_password = generate_password_hash(data['password'], method='sha256')
+
+            new_user = Student(
+                            name=data['full_name'],
+                            public_id=str(uuid.uuid4()),
+                            email=data['email'],
+                            country_name=data['country'],
+                            phone_number=data['phone'],
+                            profile_picture=data['profile_pic'],
+                            birth_date=data['birth_date'],
+                            gender=data['gender'],
+                            password=hashed_password,
+                            registeration_date=data['registeration_date'],
+                            type="student"
+                            )
+
+            db.session.add(new_user)
+            db.session.commit()
+            logger.info('user succeeded to register')
+            return jsonify({'status': 'created'}), 200
+
+        except:
+            logger.exception('user faild to register')
+            return jsonify({'message': 'Faild to register'}), 500
+
+    else:
+        logger.error('Invalid input action ')
+        return jsonify({'message': 'Invalid input action '}), 500
+
+
+        # checking if user exists or not 
+        user_email = data['email']
+        user = User.query.filter_by(email=user_email).first()
+        if user is not None:
+            return jsonify({"status": "1"})
+
+        user_phone = data['phone']
+        user = User.query.filter_by(phone_number=user_phone).first()
+        if user is not None:
+            return jsonify({"status": "2"})
+
+        hashed_password = generate_password_hash(data['password'], method='sha256')
+
+        new_user = Student(
+            name=data['full_name'],
+            public_id=str(uuid.uuid4()),
+            email=data['email'],
+            country_name=data['country'],
+            phone_number=data['phone'],
+            profile_picture=data['profile_pic'],
+            birth_date=data['birth_date'],
+            gender=data['gender'],
+            password=hashed_password,
+            registeration_date=data['registeration_date'],
+            type="student"
+        )
+
+        db.session.add(new_user)
+        db.session.commit()
+        return jsonify({'status': 'created'}), 200
+
```

Comment:
this function has been changed! please revise it with @AhmedElsayed101

Reply:
I just add try and except to check invaild queries
and check if     data["action"]  not invaild